source/mainnet/net/references/grpc.rst

```diff
@@ -0,0 +1,886 @@
+
+   :param blockHash: The block to get ancestors of.
+   :type blockHash: |grpc-block-hash|_
+   :param UInt64 amount: The requested number of ancestors.
+   :returns: A list of block hashes.
+   :rtype: ``[BlockHash]`` (see JSON schema below)
+
+   .. collapse:: View JSON schema
+
+      .. literalinclude:: grpc-json-schemas/GetAncestors.json
+         :language: json
+
+.. function:: GetBranches() -> Branch
+
+   Get the branches of the tree. This is the part of the tree above the last
+   finalized block.
+
+   :returns: The branches of the tree.
+   :rtype: ``Branch`` (see JSON schema below)
+
+   .. collapse:: View JSON schema
+
+      .. literalinclude:: grpc-json-schemas/GetBranches.json
+         :language: json
+
+Accounts
+========
+
+.. function:: GetAccountList(blockHash) -> ?[AccountAddress]
+
+   Get a list of all accounts that existed when the given block was created.
+
+   :param blockHash: The given block.
+   :type blockHash: |grpc-block-hash|_
+   :returns: A list of accounts.
+   :rtype: ``AccountList`` (see JSON schema below)
+
+   .. collapse:: View JSON schema
+
+      .. literalinclude:: grpc-json-schemas/GetAccountList.json
+         :language: json
+
+
+.. function:: GetAccountInfo(blockHash, accountAddress) -> ?AccountInfo
+
+   Get the state of an account in the given block.
+
+   :param blockHash: The given block.
+   :param accountAddress: The account to query.
+   :type blockHash: |grpc-block-hash|_
+   :type accountAddress: |grpc-account-address|_
+   :returns: The state of the account.
+   :rtype: ``?AccountInfo`` (see JSON schema below)
+
+   .. collapse:: View JSON schema
+
+      .. literalinclude:: grpc-json-schemas/GetAccountInfo.json
+         :language: json
+
+
+.. function:: GetAccountNonFinalizedTransactions(accountAddress) -> ?[TransactionHash]
+
+   Get a list of non-finalized transactions present on an account.
+
+   :param accountAddress: The account to query.
+   :type accountAddress: |grpc-account-address|_
+   :returns: A list of hashes of non-finalized transactions.
+   :rtype: ``[TransactionHash]`` (see JSON schema below)
+
+   .. collapse:: View JSON schema
+
+      .. literalinclude:: grpc-json-schemas/GetAccountNonFinalized.json
+         :language: json
+
+.. function:: GetNextAccountNonce(accountAddress) -> ?AccountNonceResponse
+
+   Returns the next available nonce for this account.
+
+   :param accountAddress: The account to query.
+   :type accountAddress: |grpc-account-address|_
+   :returns: An account nonce and whether there are any non-finalized
+             transactions for the account.
+   :rtype: ``?AccountNonceResponse`` (see JSON schema below)
+
+   .. collapse:: View JSON schema
+
+      .. literalinclude:: grpc-json-schemas/GetNextAccountNonce.json
+         :language: json
+
+Smart contracts
+===============
+
+.. function:: GetModuleList(blockHash) -> ?[ModuleHash]
+
+   Get a list of all smart contract modules that existed when the given block was created.
```

Comment:
The phrasing "when the given block was created" is a bit too imprecise.

I would rephrase as 
```
   Get a list of all smart contract modules that exist in the state at the end of the given block.
```
and the same below.